tests/test_helpers.py

```diff
@@ -515,6 +515,77 @@ def test_proxies_from_env_http_with_auth(mocker) -> None:
     assert proxy_auth.encoding == "latin1"
 
 
+# --------------------- get_env_proxy_for_url ------------------------------
+
+
+@pytest.mark.parametrize(
+    ("bypass", "proxies", "url_input", "expected_err_msg"),
+    (
+        (
+            True,
+            {"http": helpers.ProxyInfo("http://example.com", "")},
+            "http://aiohttp.io/path",
+            r"Proxying is disallowed for `'aiohttp.io'`",
+        ),
+        (
+            False,
+            {"https": helpers.ProxyInfo("https://example.com", "")},
+            "http://aiohttp.io/path",
+            r"No proxies found for `http://aiohttp.io/path` in the env",
+        ),
+        (
+            False,
+            {"https": helpers.ProxyInfo("https://example.com", "")},
+            "",
+            r"No proxies found for `` in the env",
+        ),
+    ),
+    ids=(
+        "url_matches_the_no_proxy_list",
+        "url_scheme_does_not_match_proxy_list",
+        "url_is_empty",
+    ),
+)
+def test_get_env_proxy_for_url_negative(
+    bypass,
+    expected_err_msg,
+    mocker,
+    proxies,
+    url_input,
+) -> None:
+    mocker.patch("aiohttp.helpers.proxy_bypass", return_value=bypass)
+    mocker.patch("aiohttp.helpers.proxies_from_env", return_value=proxies)
```

Comment:
@scirelli now that I look at this closer and I realize that it all depends on the env vars. Could you explore using `monkeypatch.setenv()` instead of `mocker.patch()`? https://docs.pytest.org/en/stable/monkeypatch.html

It's always better to have a full integration test rather than a unit test with everything mocked.

Reply:
Setting env vars would also demonstrate how this feature is supposed to work in real life.

Reply:
Done. 

FYI:
> WARNING  aiohttp.client:helpers.py:253 HTTPS proxies https://example.com are not supported, ignoring

Which is why all proxies are http.

Reply:
Btw, am I supposed to click "Resolve conversation" on these or do you?

Reply:
It depends. It's usually fine to resolve if you did what was asked but if there's an ongoing debate, it's better to keep it open. GH autocollapses conversations on the changed lines anyway marking them outdated. But if it's marked manually but it's not actually solved, it's harder for the reviewers to check what is the state of the PR because it adds more cognitive overhead to try to remember what was discussed compared to what was changed.

Reply:
I recently saw this nice article on reviews, you may find it interesting: https://mtlynch.io/code-review-love/.